tests/telosbookdex/test_market_ids.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import random
+import logging
+
+from pytest_eosiocdt import Asset, Symbol
+
+from .constants import telosbookdex
+
+
+def test_market_id(telosbookdex):
+    """The idea of this test is to create a lot of markets on book dex to
+    display an intermitent bug were sometimes the id of the market is not
+    divisible by two.
+
+    I think the problem is in:
+        ``vapaee::dex::market::aux_get_canonical_index_for_symbols``
```

Comment:
Yes, It was. You found an old bug thanks to this test!! Good job!

The problem is in [line 110 of market](https://github.com/guilledk/vapaee-smart-contracts/blob/c1ce2955aa859ab9f9da3fd38c04c5b264574cc8/include/vapaee/dex/modules/market.hpp#L110) module. The second "if" should be the opposite of the first but instead it's a copy-paste! So, it never enters in the case of B being currency for A in a secondary group.
```c++
// ...
else if (!A_is_currency_for_B && B_is_currency_for_A)
// ...
```